httpcore5/src/main/java/org/apache/hc/core5/util/ReflectionUtils.java

```diff
@@ -72,4 +75,35 @@ public static int determineJRELevel() {
         return 7;
     }
 
+    /**
+     * @since 5.3
+     */
+    public static <T> SocketOption<T> getExtendedSocketOptionOrNull(final String fieldName) {
+        try {
+            final Class<?> extendedSocketOptionsClass = Class.forName("jdk.net.ExtendedSocketOptions");
+            final Field field = extendedSocketOptionsClass.getField(fieldName);
+            return (SocketOption)field.get((Object)null);
+        } catch (final Exception ignore) {
+            return null;
+        }
+    }
+
+    /**
+     * object can be ServerSocket or Socket
+     *
+     * @since 5.3
+     */
+    public static <T> void setOption(final T object, final String fieldName, final T value) {
+        try {
+            final Class<?> serverSocketClass = object.getClass();
+            final Method setOptionMethod = serverSocketClass.getMethod("setOption", SocketOption.class, Object.class);
+            final SocketOption<Integer> tcpKeepIdle = getExtendedSocketOptionOrNull(fieldName);
+            if (tcpKeepIdle == null) {
```

Comment:
@kkewwei The choice of the name looks bad. Can you come up with a better name here?